File: app/src/main/java/com/github/pockethub/android/ui/item/commit/CommitFileLineItem.java
package com.github.pockethub.android.ui.item.commit;

import android.support.annotation.NonNull;
import android.view.View;
import android.widget.TextView;

import com.github.pockethub.android.R;
import com.github.pockethub.android.ui.commit.DiffStyler;
import com.github.pockethub.android.ui.item.BaseDataItem;
import com.github.pockethub.android.ui.item.BaseViewHolder;

import butterknife.BindView;

public class CommitFileLineItem extends BaseDataItem<CharSequence, CommitFileLineItem.ViewHolder> {

    private final DiffStyler diffStyler;

    public CommitFileLineItem(DiffStyler diffStyler, CharSequence line) {
        super(null, line, line.hashCode());
        this.diffStyler = diffStyler;
    }

    @Override
    public void bind(@NonNull ViewHolder viewHolder, int position) {
        viewHolder.diff.setText(getData());
        diffStyler.updateColors(getData(), viewHolder.diff);
    }

    @Override
    public int getLayout() {
        return R.layout.commit_diff_line;
    }

    @NonNull
    @Override
    public ViewHolder createViewHolder(@NonNull View itemView) {
        return new ViewHolder(itemView);
    }

    public class ViewHolder extends BaseViewHolder {

        @BindView(R.id.tv_diff) TextView diff;

        public ViewHolder(@NonNull View rootView) {
            super(rootView);
        }
    }
}

File: app/src/main/java/com/github/pockethub/android/ui/item/commit/CommitFileLineItem.kt
package com.github.pockethub.android.ui.item.commit

import com.github.pockethub.android.R
import com.github.pockethub.android.ui.commit.DiffStyler
import com.xwray.groupie.kotlinandroidextensions.Item
import com.xwray.groupie.kotlinandroidextensions.ViewHolder
import kotlinx.android.synthetic.main.commit_diff_line.*

class CommitFileLineItem(
        private val diffStyler: DiffStyler,
        val line: CharSequence
) : Item(line.hashCode().toLong()) {

    override fun getLayout() = R.layout.commit_diff_line

    override fun bind(holder: ViewHolder, position: Int) {
        holder.tv_diff.text = line
        diffStyler.updateColors(line, holder.tv_diff)
    }
}
